### ISSUE TITLE
File naming changed in Doxygen 1.9.0

### ISSUE THREAD

**Issue by swt2c** [doxygen/doxygen#8300]
**Describe the bug**
Doxygen 1.9.0 seems to be prepending an extra directory name on files with duplicate base names (for example, wx/panel.h and wx/ribbon/panel.h).

Doxygen 1.8.20 names:
```
$ ls xml.1_8_20
classwxPanel.xml
classwxRibbonPanelEvent.xml
classwxRibbonPanel.xml
combine.xslt
compound.xsd
dir_dadf652a82bb8e5033daa635a124ed82.xml
dir_f089d70a56859c57d5f3a167a2455a35.xml
index.xml
index.xsd
panel_8h.xml
ribbon_2panel_8h.xml
xml.xsd
```
Doxygen 1.9.0 names:
```
$ ls xml.1_9_0
classwxPanel.xml
classwxRibbonPanelEvent.xml
classwxRibbonPanel.xml
combine.xslt
compound.xsd
dir_dadf652a82bb8e5033daa635a124ed82.xml
dir_f089d70a56859c57d5f3a167a2455a35.xml
index.xml
index.xsd
wx_2panel_8h.xml
wx_2ribbon_2panel_8h.xml
xml.xsd
```
This seems to have changed with c3d98a0.

**To Reproduce**
Example attached.  
[example.tar.gz](https://github.com/doxygen/doxygen/files/5761050/example.tar.gz)

**Comment by doxygen** [comment]
@swt2c Please verify if the referenced commit fixes the problem.

**Comment by swt2c** [comment]
Yes, that fixes it.  Thank you once again!  :+1:

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.1.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
